app/controllers/dishes_controller.rb

```diff
@@ -0,0 +1,11 @@
+class DishesController < ApplicationController
+  def show 
+    @dish = Dish.find(params[:id])
+  end
+
+  def create 
+    # require 'pry'; binding.pry
+    DishIngredient.create!(dish_id: params[:id], ingredient_id: params[:ingredient_id])
```

Comment:
In general, we want to reserve the `dishes#create` action here for when we're creating a new dish model object. In this case, we are instead creating a relationship between an existing dish and an existing ingredient. So this is a case where a more RESTful controller design would be using a joins controller: DishIngredientsController. You could take this whole create action (the content here is great) and just plop it in a joins controller and that would be a more RESTful implementation. For a path directing to that controller, you could use "/dish_ingredients" or "/dishes/:dish_id/ingredients" (I prefer this second one because you can add the dish_id into the path)